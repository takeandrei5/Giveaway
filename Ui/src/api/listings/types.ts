import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
