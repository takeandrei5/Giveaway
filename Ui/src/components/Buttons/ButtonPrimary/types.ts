import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
