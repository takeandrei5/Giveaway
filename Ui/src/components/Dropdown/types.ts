export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
