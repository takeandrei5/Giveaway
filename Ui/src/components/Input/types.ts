export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
