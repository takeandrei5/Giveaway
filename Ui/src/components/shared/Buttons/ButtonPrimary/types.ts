import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
