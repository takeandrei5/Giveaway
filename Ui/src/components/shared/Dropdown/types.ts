export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
