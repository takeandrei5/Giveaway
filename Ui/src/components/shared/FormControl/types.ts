export type FormControlProps = {
	children: React.ReactElement;
	label: string;
	id: string;
	name: string;
};

export type useCheckFormIsInvalidProps = {
	isInvalid: boolean;
};
