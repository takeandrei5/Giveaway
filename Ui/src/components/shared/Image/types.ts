import { ImageProps as NextImageProps } from 'next/image';

export type ImageProps = NextImageProps & {
	height: string;
	width: string;
	backgroundColor?: string;
	borderRadius?: string;
	draggable?: boolean;
};
