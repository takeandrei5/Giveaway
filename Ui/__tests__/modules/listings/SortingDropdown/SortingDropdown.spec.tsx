import { SortingDropdown } from '@modules/listings/SortingDropdown';
import { SortingDropdownProps } from '@modules/listings/SortingDropdown/types';
import { render } from '@testing-library/react';

describe('SortingDropdown', () => {
	it('should match snapshot', () => {
		// Arrange
		const props: SortingDropdownProps = {
			id: 'sorting-dropdown',
			options: [],
			onChangeHandler: (value: string) => {},
			name: 'sorting-dropdown-test-name',
			value: 'sorting-dropdown-test-value',
		};

		// Act
		const { container } = render(<SortingDropdown {...props} />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
