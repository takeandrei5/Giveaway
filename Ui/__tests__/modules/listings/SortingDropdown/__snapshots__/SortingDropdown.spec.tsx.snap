// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingDropdown should match snapshot 1`] = `
<div>
  <div
    class="css-13dls4l"
  >
    <div
      class="css-13n1q9m"
    >
      <span
        class="chakra-text css-q4il9g"
        data-testid="typography"
      >
        Sort by:
      </span>
      <div
        class="chakra-select__wrapper css-yumodq"
      >
        <select
          class="chakra-select css-1iot0zm"
          data-testid="dropdown"
          id="sorting-dropdown"
          name="sorting-dropdown-test-name"
        />
        <div
          class="chakra-select__icon-wrapper css-6fsspc"
        >
          <svg
            aria-hidden="true"
            class="chakra-select__icon"
            focusable="false"
            role="presentation"
            style="width: 1em; height: 1em; color: currentColor;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
